### PR TITLE
TopAppBar: Add centered title option

### LINKED ISSUE #189: mwc-top-app-bar: can't center title
Right now the title is always left-aligned in the toolbar. Is there any way to make the title center? Something like this:

screen shot 2019-01-15 at 2 01 36 pm

### PR DESCRIPTION
**Pull request by straversi**
Allow the title to be centered on the Top App Bar. The option is exposed as a boolean property on the host called `centerTitle`. Open to suggestions on this API choice.

demo top app bar

_Note: This is not a built-in feature from the material spec._ That being said, centered titles are featured in some of the Material custom theme case studies. (see "Rally Theme" from [Top App Bar spec](https://material.io/design/components/app-bars-top.html#theming))

closes #189

### CODE REVIEW

**Comment by tionkje** [comment]
How about an
alignTitle="center"
Attribute? So it would have a default of left and leaves open later support for other types of alignment.

**Comment by straversi** [comment]
@tionkje You're right, string is a more natural type for this. I don't see any problems with that 👍

File: packages/top-app-bar/src/mwc-top-app-bar.scss [hunk 1]
```diff
@@ -25,6 +25,26 @@ limitations under the License.
...
+.mwc-top-app-bar--center-title {
+  .mdc-top-app-bar__section--align-start {
+    flex-basis: 0%;
+  }
+  .mdc-top-app-bar__section--align-end {
+    flex-basis: 0%;
+  }
+  .mdc-top-app-bar__row {
+    justify-content: center;
+  }
```
[the remark this answers is not in the document]
Comment (straversi) [reply packages/top-app-bar/src/mwc-top-app-bar.scss]: Nope, not sure why that's there. 👍

**Comment by straversi** [comment]
Alright, trying out a type: string property. Feels a little awkward in the source given that there are only two options right now, one of which is the existing "default", but the interface is nicer.

**Comment by frankiefu** [comment]
Yeah, I'd suggest to keep the property as boolean. From the Material spec, afaik the title in the app bar could either be left or center aligned.

**Comment by straversi** [comment]
Ok, and if more alignment types are ever introduced, that would be a good time to refactor this property's type to `string`.

**Pull request merged**